mathesar/views/api.py

```diff
@@ -85,11 +85,18 @@ def create(self, request):
         serializer.is_valid(raise_exception=True)
 
         valid_data = serializer.validated_data
-        if 'data_files' in valid_data and valid_data['data_files']:
-            table = create_table_from_datafile(valid_data)
-        else:
-            table = create_empty_table(valid_data)
-
+        try:
+            if 'data_files' in valid_data and valid_data['data_files']:
+                table = create_table_from_datafile(valid_data)
+            else:
+                table = create_empty_table(valid_data)
+        except ProgrammingError as e:
+            if type(e.orig) == DuplicateTable:
+                raise ValidationError(
+                    f"Table {request.data['name']} already exists"
```

Comment:
This should use "Relation" instead of table.  The error also will occur if:
- A view of that name already exists, or
- An index of that name already exists
This list is probably not complete.

Also, consider referring to the schema in the error message ("Relation <table> already exists in schema <schema>").  Table names aren't globally unique in a DB.